src/wsched/ScanScheduler.cc

```diff
@@ -221,23 +209,59 @@ util::Command::Ptr ScanScheduler::getCmd(bool wait)  {
 
 
 void ScanScheduler::queCmd(util::Command::Ptr const& cmd) {
-    wbase::Task::Ptr t = std::dynamic_pointer_cast<wbase::Task>(cmd);
-    if (t == nullptr) {
-        LOGS(_log, LOG_LVL_WARN, getName() << " queCmd could not be converted to Task or was nullptr");
-        return;
+    vector<util::Command::Ptr> vect;
+    vect.push_back(cmd);
+    queCmd(vect);
+}
+
+
+void ScanScheduler::queCmd(vector<util::Command::Ptr> const& cmds) {
+    LOGS(_log, LOG_LVL_TRACE, "ScanScheduler::queCmd cmds.sz=" << cmds.size());
+    std::vector<wbase::Task::Ptr> tasks;
+    bool first = true;
+    QueryId qid;
+    int jid = 0;
+    // Convert to a vector of tasks
+    for (auto const& cmd:cmds) {
+        wbase::Task::Ptr t = dynamic_pointer_cast<wbase::Task>(cmd);
+        if (t == nullptr) {
+            LOGS(_log, LOG_LVL_ERROR, getName()
+                    << " queCmd could not be converted to Task or was nullptr");
+            return;
```

Comment:
Why return, there are other tasks to process? If this is a condition that should never happen then throwing an exception is probably better?